test/src/test/java/org/jenkins/ui/symbol/SymbolJenkinsTest.java

```diff
@@ -0,0 +1,46 @@
+package org.jenkins.ui.symbol;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.not;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.jupiter.api.DisplayName;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.RealJenkinsRule;
+
+public class SymbolJenkinsTest {
+    @Rule
+    public RealJenkinsRule rjr = new RealJenkinsRule().addPlugins("plugins/design-library.jpi");
+
+    @Test
+    @DisplayName("When resolving a symbol from a missing plugin, the placeholder is generated instead")
+    public void missingSymbolFromPluginDefaultsToPlaceholder() throws Throwable {
+        rjr.then(SymbolJenkinsTest::_missingSymbolFromPluginDefaultsToPlaceholder);
+    }
+
+    private static void _missingSymbolFromPluginDefaultsToPlaceholder(JenkinsRule j) {
+        String symbol = Symbol.get(new SymbolRequest.Builder()
+                                           .withName("science")
+                                           .withPluginName("missing-plugin")
```

Comment:
This would not work out of the box in my use case. I have a given symbol as full referenced string with plugin included, e.g. `"symbol-solid/triangle-exclamation plugin-font-awesome-api"`. I can split this manually before and after the space but it would be simpler if another builder method would be available that takes the whole string. What do you think, does this make sense?

Reply:
I tried to use the new API in https://github.com/jenkinsci/prism-api-plugin/pull/60. However, it seems that I only get a cross as icon. It would be really helpful if we can add some logging why an icon is not found :-(

Or do I need to remove the `plugin-` suffix?

Reply:
Ok, I got it working now:

```
             String[] elements = StringUtils.split(name);
             if (elements.length == 2) {
                 String symbol = Symbol.get(new SymbolRequest.Builder()
                         .withName(StringUtils.removeStart(elements[0], "symbol-"))
                         .withPluginName(StringUtils.removeStart(elements[1], "plugin-"))
                         .withClasses(ICON_MD)
                         .build());
                 return new UnescapedText(symbol);
             }
```

From my perspective it would be helpful, if there would be another builder method (something like `withId` or `withIdentifier`), that takes the whole symbol identifier and splits it accordingly.